src/topology/algebra/group.lean

```diff
@@ -263,15 +263,19 @@ protected def homeomorph.inv (G : Type*) [topological_space G] [has_involutive_i
 @[to_additive] lemma is_closed_map_inv : is_closed_map (has_inv.inv : G → G) :=
 (homeomorph.inv _).is_closed_map
 
-variables {G}
-
-@[to_additive] lemma is_open.inv (hs : is_open s) : is_open s⁻¹ := hs.preimage continuous_inv
-@[to_additive] lemma is_closed.inv (hs : is_closed s) : is_closed s⁻¹ := hs.preimage continuous_inv
 @[to_additive] lemma inv_closure : ∀ s : set G, (closure s)⁻¹ = closure s⁻¹ :=
 (homeomorph.inv G).preimage_closure
 
 end continuous_involutive_inv
 
+section continuous_inv
+variables {G} [topological_space G] [has_inv G] [has_continuous_inv G] {s : set G}
+
+@[to_additive] lemma is_open.inv (hs : is_open s) : is_open s⁻¹ := hs.preimage continuous_inv
+@[to_additive] lemma is_closed.inv (hs : is_closed s) : is_closed s⁻¹ := hs.preimage continuous_inv
+
+end continuous_inv
```

Comment:
I'm wary of this generalization'; the definition of `set.has_inv` is in my opinion just wrong on non-involutive inverses; I made a zulip thread about this before.

Reply:
Ok great, I'll remove this one, thanks!